autopilot/sagemaker_autopilot_direct_marketing.ipynb

```diff
@@ -0,0 +1,647 @@
+    "        TransformJobName=transformer._current_job_name\n",
+    "    )\n",
+    "    job_run_status = describe_response[\"TransformJobStatus\"]\n",
+    "    print(job_run_status)\n",
+    "    if job_run_status in (\"Failed\", \"Completed\", \"Stopped\"):\n",
+    "        break"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now let's view the results of the transform job:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "test_data_preds = pd.read_csv(\n",
+    "    transformer.output_path + \"test_data.csv.out\",\n",
+    "    header=None,\n",
+    "    names=[\"predicted_label\", \"probability\"],\n",
+    ")\n",
+    "\n",
+    "test_data_preds"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can use the result of the transform job to evaluate additional metrics on the test dataset."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sklearn.metrics import (\n",
+    "    average_precision_score,\n",
+    "    classification_report,\n",
+    "    f1_score,\n",
+    "    precision_recall_curve,\n",
+    "    roc_auc_score,\n",
+    "    roc_curve,\n",
+    ")\n",
+    "\n",
+    "labels = test_data[\"y\"]\n",
+    "AUC = roc_auc_score(labels == \"yes\", test_data_preds.probability)\n",
+    "AP = average_precision_score(labels, test_data_preds.probability, pos_label=\"yes\")\n",
+    "F1 = f1_score(labels, test_data_preds.predicted_label, pos_label=\"yes\")\n",
+    "\n",
+    "print(f\"AUC: {AUC:.3f}\\nAP {AP:.3f}\\nF1 {F1:.3f}\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(classification_report(labels == \"yes\", test_data_preds.predicted_label == \"yes\"))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import matplotlib.pyplot as plt\n",
+    "\n",
+    "plt.figure(num=None, figsize=(16, 9), dpi=160, facecolor=\"w\", edgecolor=\"k\")\n",
+    "fpr, tpr, _ = roc_curve(labels == \"yes\", test_data_preds.probability)\n",
+    "plt.plot(fpr, tpr)\n",
+    "plt.plot([0, 1], [0, 1])\n",
+    "plt.xlabel(\"False Positive Rate\")\n",
+    "plt.ylabel(\"True Positive Rate\")\n",
+    "plt.title(\"ROC Curve\")\n",
+    "\n",
+    "\n",
+    "plt.figure(num=None, figsize=(16, 9), dpi=160, facecolor=\"w\", edgecolor=\"k\")\n",
+    "precision, recall, _ = precision_recall_curve(\n",
+    "    labels == \"yes\", test_data_preds.probability\n",
+    ")\n",
+    "plt.plot(recall, precision)\n",
+    "plt.plot([0, 1], [1, 0])\n",
+    "plt.xlabel(\"Recall\")\n",
+    "plt.ylabel(\"Precision\")\n",
+    "plt.title(\"Precision-Recall Curve\")\n",
+    "plt.title(\"ROC Curve\");"
```

Comment:
Delete this extra line - overwrites the `Precision-Recall Curve` title just above

Reply:
Done